soda_machine.rb

```diff
@@ -7,12 +7,20 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.size
   end
 
   def find_soda(soda_brand)
+    @sodas.find{|soda| soda.brand == soda_brand}
   end
 
   def sell(soda_brand)
+    this_soda = find_soda(soda_brand)
+    if this_soda != nil
+      @sodas.delete_at(@sodas.index(this_soda))
```

Comment:
This is a nice addition to the test suite, but I just wanted to point out that @amgando 's suggested change would still work given the new test! Because `#delete` checks for equality, and `pespi != second_pepsi` in your test (they are different objects).  

It would be pretty weird to have the same soda object in your machine twice -- you might have multiple cans of the same *kind* of soda, but never multiple of the exact same can; that wouldn't model the real world correctly.